File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task52_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <stdlib.h>

#define KEY 0x42

void encrypt_decrypt(const char* input_file, const char* output_file) {
    FILE *fin, *fout;
    int ch;

    fin = fopen(input_file, "rb");
    if (fin == NULL) {
        printf("Error opening input file.\
");
        exit(1);
    }

    fout = fopen(output_file, "wb");
    if (fout == NULL) {
        printf("Error opening output file.\
");
        fclose(fin);
        exit(1);
    }

    while ((ch = fgetc(fin)) != EOF) {
        fputc(ch ^ KEY, fout);
    }

    fclose(fin);
    fclose(fout);
}

int main() {
    char choice;
    char input_file[100], output_file[100];

    printf("Enter \'e\' to encrypt or \'d\' to decrypt: ");
    scanf(" %c", &choice);
    printf("Enter input file name: ");
    scanf("%s", input_file);
    printf("Enter output file name: ");
    scanf("%s", output_file);

    encrypt_decrypt(input_file, output_file);
    printf("Operation completed successfully.\
");

    return 0;
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task52_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.*;
import java.util.Scanner;

public class FileEncryptDecrypt {
    private static final byte KEY = 0x42;

    public static void encryptDecrypt(String inputFile, String outputFile) throws IOException {
        try (FileInputStream fis = new FileInputStream(inputFile);
             FileOutputStream fos = new FileOutputStream(outputFile)) {
            int data;
            while ((data = fis.read()) != -1) {
                fos.write(data ^ KEY);
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter \'e\' to encrypt or \'d\' to decrypt: ");
        String choice = scanner.nextLine();
        System.out.print("Enter input file name: ");
        String inputFile = scanner.nextLine();
        System.out.print("Enter output file name: ");
        String outputFile = scanner.nextLine();

        try {
            encryptDecrypt(inputFile, outputFile);
            System.out.println("Operation completed successfully.");
        } catch (IOException e) {
            System.out.println("An error occurred: " + e.getMessage());
        }
    }
}
